src/utils/bridgeMonitor.js

```diff
@@ -0,0 +1,169 @@
+  return undefined;
+};
+
+const updateMilestoneStatusToPaid = async (app, donation) => {
+  const donationService = app.service('donations');
+  const milestoneService = app.service('milestones');
+  const milestone = await milestoneService.get(donation.ownerTypeId);
+  const { maxAmount, reviewerAddress, fullyFunded } = milestone;
+
+  // never set uncapped or without-reviewer non-fullyFunded milestones as PAID
+  const hasReviewer = reviewerAddress && reviewerAddress !== ZERO_ADDRESS;
+  if (!maxAmount || (!fullyFunded && !hasReviewer)) return;
+  const donations = await donationService.find({
+    paginate: false,
+    query: {
+      ownerTypeId: donation.ownerTypeId,
+      status: { $in: [DonationStatus.COMMITTED, DonationStatus.PAYING, DonationStatus.PAID] },
+      amountRemaining: { $ne: '0' },
+    },
+  });
+
+  // if there are still committed and paying donations, don't mark the as paid or paying
+  if (
+    donations.some(d => d.status === DonationStatus.COMMITTED || d.status === DonationStatus.PAYING)
+  ) {
+    return;
+  }
+
+  const hasDonationWithPendingStatusInBridge = donations.some(
+    d =>
+      d.bridgeStatus !== DONATION_BRIDGE_STATUS.PAID &&
+      d.bridgeStatus !== DONATION_BRIDGE_STATUS.CANCELLED,
+  );
+  if (!hasDonationWithPendingStatusInBridge) {
+    await app.service('milestones').patch(donation.ownerTypeId, {
+      isAllDonationsPaidInBridge: true,
+    });
+    // TODO can email to milestone owner and says that the money went to their wallet
+  }
+};
+
+const updateDonationsAndMilestoneStatusToBridgePaid = async ({ app, donation, bridgeEvent }) => {
+  const donationService = app.service('donations');
+  const bridgeStatus = DONATION_BRIDGE_STATUS.PAID;
+  await donationService.patch(donation._id, {
+    bridgeStatus,
+    bridgeTxHash: bridgeEvent.transactionHash,
+  });
+  logger.info('update donation bridge status', {
+    donationId: donation._id,
+    bridgeStatus,
+  });
+  if (donation.ownerType === AdminTypes.MILESTONE) {
+    await updateMilestoneStatusToPaid(app, donation);
+  }
+};
+const updateDonationsAndMilestoneStatusToBridgeFailed = async ({ app, donation, bridgeEvent }) => {
+  const donationService = app.service('donations');
+  const bridgeStatus = DONATION_BRIDGE_STATUS.CANCELLED;
+  await donationService.patch(donation._id, {
+    bridgeStatus,
+    bridgeTxHash: bridgeEvent.transactionHash,
+  });
+  logger.info('update donation bridge status', {
+    donationId: donation._id,
+    bridgeStatus,
+  });
+  if (donation.ownerType === AdminTypes.MILESTONE) {
+    // TODO I dont know what should to do with milestone in this case
+  }
+};
+const updateDonationsAndMilestoneStatusToBridgeUnknown = async ({ app, donation }) => {
+  const donationService = app.service('donations');
+  let bridgeStatus = DONATION_BRIDGE_STATUS.UNKNOWN;
+  const timeBetweenCreatedDonationAndNow =
+    new Date().getTime() - new Date(donation.createdAt).getTime();
+  const threshold = 60 * 24 * 3600 * 1000;
```

Comment:
Improve naming and add some comment to explain the logic

Reply:
done